### PR TITLE
Support for JDK 11

### PR DESCRIPTION
**Pull request by pierre**
See individual commits. CI likely won't pass until Kill Bill is updated.

Note the new `javac-release` profile automatically enabled for JDK11+ builds.

/cc @andrenpaes

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -426,12 +432,14 @@
             <dependency>
                 <groupId>io.airlift</groupId>
                 <artifactId>concurrent</artifactId>
-                <version>202</version>
+                <!-- 207 requires Java 11 -->
+                <version>206</version>
```
Comment (sbrossie) [inline pom.xml:436]: Would it make sense to have a version that is different based on the target java version we are compiling for?

File: pom.xml [hunk 2]
```diff
@@ -2677,5 +2692,16 @@
                 </plugins>
             </build>
         </profile>
+        <profile>
+            <id>javac-release</id>
+            <activation>
+                <jdk>[11,)</jdk>
```
Comment (sbrossie) [inline pom.xml:2698]: What does this mean exactly - in plain english?

**Review by sbrossie — APPROVED** [review]
Seems painful... 👏

File: pom.xml (hunk 1 above)
Comment (pierre) [reply pom.xml:436]: Currently, we only produce builds that are binary compatible with 1.8. I would suggest to drop 1.8 compatibility and switch to 11 compatibility as part of our 0.24.x release.

File: pom.xml (hunk 2 above)
Comment (pierre) [reply pom.xml:2698]: _When building using JDK 11 or greater, pass `--release 8` to `javac` and set `--illegal-access=permit` when running tests._
